internal/scorecard/plugins/plugin_runner.go

```diff
@@ -130,9 +132,27 @@ func RunInternalPlugin(pluginType PluginType, config BasicAndOLMPluginConfig, lo
 		if err != nil {
 			return scapiv1alpha1.ScorecardOutput{}, fmt.Errorf("failed to read csv: %v", err)
 		}
+
 		if err = yaml.Unmarshal(yamlSpec, csv); err != nil {
 			return scapiv1alpha1.ScorecardOutput{}, fmt.Errorf("error getting ClusterServiceVersion: %v", err)
 		}
+
+		csvValidator := validation.ClusterServiceVersionValidator
+		results := csvValidator.Validate(csv)
+		for _, r := range results {
+			if r.HasError() {
+				var errorMsgs strings.Builder
+				for _, e := range r.Errors {
+					errorMsgs.WriteString(e.Error())
+				}
+				return scapiv1alpha1.ScorecardOutput{}, fmt.Errorf("error validating ClusterServiceVersion: %s", errorMsgs.String())
+			}
+			if r.HasWarn() {
+				for _, w := range r.Warnings {
+					log.Warnf("CSV validation warning: type [%s] %s", w.Type, w.Detail)
+				}
+			}
+		}
```

Comment:
Curious why we need this and the CSVValidator test?

Reply:
I was interpreting the previous code to be doing a semi-validation of the CSV using the Unmarshal, so I added this validation logic here to provide a more comprehensive validation of the CSV.  From before, logic below that uses the CSV to pull out CRs if OLMDeployed is set.

Reply:
yep, I think you are right, if we are going to validate it up front, there is no need for a csv validation test, I'll remove it as a test.